src/attacks.rs

```diff
@@ -115,17 +129,31 @@ mod test {
             .into_iter()
             .enumerate()
             .for_each(|(i, edges)| match i {
-                63 => {
-                    let exp = vec![Edge(0, 1), Edge(2, 3), Edge(4, 5), Edge(6, 7)];
-                    assert_eq!(edges, exp);
+                0 => {
+                    assert_eq!(
+                        edges,
+                        HashSet::from_iter(
```

Comment:
Re about `HashSet`: not sure the uniqueness property is worth the loss of readability here `HashSet::from_iter` + `iter()` + `cloned()` .. ?

Reply:
This can be wrapped into a function. Sadly (or not) Rust is supposed to be _this_ verbose, I wouldn't change a feature of the code just to try to bypass that.